src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs

```diff
@@ -1135,6 +1135,168 @@ public Task ResponseTrailers_NullValues_Ignored(HttpContext context)
+            }
+        }
+        public async Task Reset_AfterResponseHeaders_Resets_Complete(HttpContext httpContext)
+        {
+            await _Reset_AfterResponseHeaders_ResetsCt.Task;
+        }
+
+        private TaskCompletionSource<object> _Reset_DuringResponseBody_ResetsCt = new TaskCompletionSource<object>();
+
+        public async Task Reset_DuringResponseBody_Resets(HttpContext httpContext)
+        {
+            try
+            {
+                Assert.Equal("HTTP/2", httpContext.Request.Protocol);
+                var feature = httpContext.Features.Get<IHttpResetFeature>();
+                Assert.NotNull(feature);
+                await httpContext.Response.WriteAsync("Hello World");
+                await httpContext.Response.Body.FlushAsync();
+                feature.Reset(1111); // Custom
+                _Reset_DuringResponseBody_ResetsCt.SetResult(0);
+            }
+            catch (Exception ex)
+            {
+                _Reset_DuringResponseBody_ResetsCt.SetException(ex);
+            }
+        }
+
+        public async Task Reset_DuringResponseBody_Resets_Complete(HttpContext httpContext)
+        {
+            await _Reset_DuringResponseBody_ResetsCt.Task;
+        }
+
+        private TaskCompletionSource<object> _Reset_BeforeRequestBody_ResetsCt = new TaskCompletionSource<object>();
+
+        public async Task Reset_BeforeRequestBody_Resets(HttpContext httpContext)
+        {
+            try
+            {
+                Assert.Equal("HTTP/2", httpContext.Request.Protocol);
+                var feature = httpContext.Features.Get<IHttpResetFeature>();
+                Assert.NotNull(feature);
+                var readTask = httpContext.Request.Body.ReadAsync(new byte[10], 0, 10);
+
+                feature.Reset(1111);
+
+                await Assert.ThrowsAsync<IOException>(() => readTask);
+
+                _Reset_BeforeRequestBody_ResetsCt.SetResult(0);
+            }
+            catch (Exception ex)
+            {
+                _Reset_BeforeRequestBody_ResetsCt.SetException(ex);
+            }
+        }
+
+        public async Task Reset_BeforeRequestBody_Resets_Complete(HttpContext httpContext)
+        {
+            await _Reset_BeforeRequestBody_ResetsCt.Task;
+        }
+
+        private TaskCompletionSource<object> _Reset_DuringRequestBody_ResetsCt = new TaskCompletionSource<object>();
+
+        public async Task Reset_DuringRequestBody_Resets(HttpContext httpContext)
+        {
+            try
+            {
+                Assert.Equal("HTTP/2", httpContext.Request.Protocol);
+                var feature = httpContext.Features.Get<IHttpResetFeature>();
+                Assert.NotNull(feature);
+
+                var read = await httpContext.Request.Body.ReadAsync(new byte[10], 0, 10);
+                Assert.Equal(10, read);
+
+                var readTask = httpContext.Request.Body.ReadAsync(new byte[10], 0, 10);
+                feature.Reset(1111);
```

Comment:
Add a test that `Reset(0)` works. Initial HttpSys implementation incorrectly didn't allow it. I want to double check it works